src/ol/renderer/canvas/TileLayer.js

```diff
@@ -841,21 +843,25 @@ class CanvasTileLayerRenderer extends CanvasLayerRenderer {
     }
     context.imageSmoothingEnabled = true;
 
-    /**
-     * Here we unconditionally expire the source cache since the renderer maintains
-     * its own cache.
-     * @param {import("../../Map.js").default} map Map.
-     * @param {import("../../Map.js").FrameState} frameState Frame state.
-     */
-    const postRenderFunction = (map, frameState) => {
-      const tileSourceKey = getUid(tileSource);
-      const wantedTiles = frameState.wantedTiles[tileSourceKey];
-      const tilesCount = wantedTiles ? Object.keys(wantedTiles).length : 0;
-      this.updateCacheSize(tilesCount);
-      this.tileCache_.expireCache();
-    };
-
-    frameState.postRenderFunctions.push(postRenderFunction);
+    if (this.renderComplete) {
+      /**
+       * Here we unconditionally expire the source cache since the renderer maintains
+       * its own cache.
+       * @param {import("../../Map.js").default} map Map.
+       * @param {import("../../Map.js").FrameState} frameState Frame state.
+       */
+      const postRenderFunction = (map, frameState) => {
+        const tileSourceKey = getUid(tileSource);
+        const wantedTiles = frameState.wantedTiles[tileSourceKey];
+        const tilesCount = wantedTiles ? Object.keys(wantedTiles).length : 0;
+        this.updateCacheSize(tilesCount);
+        this.tileCache_.expireCache();
+      };
+
+      frameState.postRenderFunctions.push(postRenderFunction);
+    } else {
+      frameState.animate = true;
```

Comment:
This fixes an issue with the new canvas tilelayer renderer. It can be seen in all our examples, e.g. https://openlayers.org/en/latest/examples/simple.html: Zoom in several times to a region with many labels on the map. Wait until the map is loaded. Then zoom out once using the '-' Button. You'll see that higher zoom tiles are rendered, which go away as soon as you start panning the map. The expected behavior would be to see the correct zoom tiles already after zooming out.